application/event_processor/dos.py

```diff
@@ -1,24 +1,21 @@
+from os import environ
 from logging import getLogger
-from os import environ, getenv
-from typing import List
-
-from boto3 import client
-from change_request import (
-    ADDRESS_CHANGE_KEY,
-    PHONE_CHANGE_KEY,
-    POSTCODE_CHANGE_KEY,
-    PUBLICNAME_CHANGE_KEY,
-    WEBSITE_CHANGE_KEY,
-)
-from nhs import NHSEntity
-from psycopg2 import connect
+from typing import List, Dict
+from datetime import datetime, date, time
+from itertools import groupby
+
+import psycopg2
```

Comment:
Can this be imported like this
```suggestion 
from psycopg2 import connect
```

Reply:
The use of explicitly stating the module the function 'connect' comes from is for readability.

`connect()`
could be used for a million different things when read in code
`psycopg2.connect()`
When reading this I can immediately see the context of the function and its clear what library is used.

Reply:
If all other variables and functions are named correctly and have descriptions that shouldn't be an issue

Reply:
Im not too precious either way here, a quick google suggests both are acceptable, we should definitely agree a convention though otherwise we will end up in a cycle of changing each others code back and forth. I can see the argument for including the library name at call time rather than import for 3rd party libraries especially when dependent on different libraries with similar exports. The boto3 is pretty horrible for this too, firstly why boto3 and the `from boto3 import client `and then having to pass loads of string literals for the service you need is horrid, whats wrong with `from awssdk import DynamoDBClient` Suggest further discussion within the team especially from those with more years of python development experience than I :)

Reply:
Me and Jack had discussed our views privately but in general the 2 main reasons you may want to explicitly state the module for the call are 
- If the function name is vague, the module adds context. 
`json.dumps()` vs `dumps()` 
or 
`psycopg2.connect()` vs `connect()` 
make it more clear the the reader what the function does. (ie. dumps a json, connects via psycopg2) and differentiates it from other modules which have their own `connect()` functions.
- It differentiates small function names from python built-in functions. 
`connect() and dumps()` 
as they are very simple one worded function names may get confused as a python built in function such as 
`sorted() enumerate() revered() slice() map()`

This will always be a grey area exactly when to do one or the other, but I think its important to take these into account for readability when writing code and making those decisions.